packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -63,3 +71,65 @@ function relativeOffset(ancestor: HTMLElement, child: HTMLElement, axis: 'left'|
   }
   return sum;
 }
+
+
+function optionsToString(options: ScrollIntoViewOptions) {
+  return Object.keys(options).sort().map(key => `${key}${options[key]}`).toString();
+}
+
+export function scrollIntoViewFully(target: Element, scrollOptions?: ScrollIntoViewOptions) {
+  const intersectionObserverOptions: IntersectionObserverInit = {
+    root: undefined,
+    rootMargin: '0px',
+    threshold: 1
+  };
+
+  if (scrollOptions == null) {
+    scrollOptions = {
+      block: 'nearest'
+    };
+  }
+
+  let intersectionObserverCallback = (entries: Array<IntersectionObserverEntry>, observer: IntersectionObserver) => {
+    entries.forEach(entry => {
+      if (!entry.isIntersecting) {
+        scrollIntoViewHelper(entry.target, scrollOptions);
+      }
+      observer.unobserve(entry.target);
+    });
+  };
+
+  if (target) {
+    let observerMapKey = optionsToString(scrollOptions);
+    let observer = observerMap.get(observerMapKey);
+    if (!observer) {
+      observer = new IntersectionObserver(intersectionObserverCallback, intersectionObserverOptions);
+      observerMap.set(observerMapKey, observer);
+    }
+
+    observer.observe(target);
+  }
+}
+
+// TODO: rename? combine with scrollintoview above? Replace scrollIntoView above (would need to add param for scrollRef so that we could have old behavior)?
+// Perhaps add a parameter to this func to customize the root so we can say how far up we want to actually adjust the scroll?
+// scrollIntoView is exported and available from aria/utils so is it a breaking change to replace it with this func?
+// TODO: test with zoom/pinch zoom
+function scrollIntoViewHelper(target: Element, scrollOptions: ScrollIntoViewOptions) {
+  let root = document.scrollingElement || document.documentElement;
+  let scrollParent = getScrollParent(target);
+
+  // If scrolling is not currently prevented then we aren’t in a overlay nor is a overlay open, just use element.scrollIntoView to bring the element into view
+  if (!isScrollPrevented) {
+    target?.scrollIntoView?.(scrollOptions);
+  } else {
+    // If scrolling is prevented, we don't want to scroll the body since it will break the open overlay's positioning.
+    while (target && scrollParent && target !== root && scrollParent !== root) {
+      // TODO: adapt scrollIntoView so that it can work with the provided ScrollIntoViewOptions instead of mimicking scrollIntoView({block: 'nearest'}) only
+      // do this if we decide we actually want the scrollOption flexibility
+      scrollIntoView(scrollParent as HTMLElement, target as HTMLElement);
```

Comment:
Open to opinions here, do we feel it is valuable to allow `ScrollIntoViewOptions` to be passed to `scrollIntoViewFully`? IMO I like it for the slider experience commented above. 

This is perhaps a good argument to use something like https://scroll-into-view-if-needed.netlify.app/ to polyfill `scrollIntoViewIfNeeded` so that we can simplify all of this code since it allows us to customize the scroll options AND define scroll boundaries/propagation (aka prevent document body from scrolling if usePreventScroll is active).

Reply:
Update: polyfill doesn't work with RTL so I'll work on modifying scrollIntoView in a followup PR when making all elements scroll into view on keyboard interaction if necessary